native/src/main/java/io/ballerina/stdlib/http/api/client/actions/HttpClientAction.java

```diff
@@ -293,22 +294,49 @@ private static String[] getPathStringArray(BArray pathArray) {
 
     private static String constructQueryString(BMap params) {
         List<String> queryParams = new ArrayList<>();
+        Map<String, String> annotationValues = getQueryNameMapping(params);
         BString[] keys = (BString[]) params.getKeys();
         if (keys.length == 0) {
             return "";
         }
         for (BString key : keys) {
             Object value = params.get(key);
+            String queryName = key.getValue();
+            queryName = annotationValues.getOrDefault(queryName, queryName);
             String valueString = value.toString();
             if (value instanceof BArray) {
                 valueString = valueString.substring(1, valueString.length() - 1);
                 valueString = valueString.replace(QUOTATION_MARK, EMPTY);
             }
-            queryParams.add(key.getValue() + EQUAL_SIGN + valueString);
+            queryParams.add(queryName + EQUAL_SIGN + valueString);
         }
         return String.join(AND_SIGN, queryParams);
     }
 
+    /**
+     * This util function extracts the query name with the query annotation.
+     *
+     * @param params - Parameter map
+     * @return Map of string with overridden query param names
+     */
+    private static Map<String, String> getQueryNameMapping(BMap params) {
+        RecordType queryRecord = (RecordType) params.getType();
+        BMap<BString, Object> annotations = queryRecord.getAnnotations();
+        Map<String, String> annotationValues = new HashMap<>();
+
+        for (Map.Entry<BString, Object> annotation: annotations.entrySet()) {
+            //store query parameters real name : $field$.<query name>
+            BString key = annotation.getKey().substring(8, annotation.getKey().length());
```

Comment:
Better to split it with the regex(starts with `$field$.`) rather than specific length, then it will be easier to change if the lang did any changes in the internal representation.

Reply:
Addressed via https://github.com/ballerina-platform/module-ballerina-http/pull/2143/commits/93a28143565d062f1a2a34e36bbfda35ea6e22a4